SDWebImage/Core/SDDiskCache.m

```diff
@@ -240,18 +242,41 @@ - (nullable NSString *)cachePathForKey:(NSString *)key {
 
 - (NSUInteger)totalSize {
     NSUInteger size = 0;
-    NSDirectoryEnumerator *fileEnumerator = [self.fileManager enumeratorAtPath:self.diskCachePath];
-    for (NSString *fileName in fileEnumerator) {
-        NSString *filePath = [self.diskCachePath stringByAppendingPathComponent:fileName];
-        NSDictionary<NSString *, id> *attrs = [self.fileManager attributesOfItemAtPath:filePath error:nil];
-        size += [attrs fileSize];
+
+    // Use URL-based enumerator instead of Path(NSString *)-based enumerator to reduce
+    // those objects(ex. NSPathStore2/_NSCFString/NSConcreteData) created during traversal.
+    // Even worse, those objects are added into AutoreleasePool, in background threads, 
+    // the time to release those objects is undifined(according to the usage of CPU)
+    // It will truely consumes a lot of VM, up to cause OOMs. 
+    NSDirectoryEnumerationOptions options = (NSDirectoryEnumerationOptions)0;
+    if (@available (iOS 13, *)) {
+        // NSDirectoryEnumerationProducesRelativePathURLs causes the NSDirectoryEnumerator 
+        // to always produce file path URLs relative to the directoryURL. This can reduce 
+        // the size of each URL object returned during enumeration.
+        options |= NSDirectoryEnumerationProducesRelativePathURLs;
+    }
+    
+    @autoreleasepool {
+        NSURL *pathURL = [NSURL fileURLWithPath:self.diskCachePath isDirectory:YES];
+        NSDirectoryEnumerator<NSURL *> *fileEnumerator = [self.fileManager enumeratorAtURL:pathURL
+                                                  includingPropertiesForKeys:@[NSURLFileSizeKey]
+                                                                     options:options
+                                                                errorHandler:NULL];
+        
+        for (NSURL *fileURL in fileEnumerator) {
+            @autoreleasepool {
+                NSNumber *fileSize;
+                [fileURL getResourceValue:&fileSize forKey:NSURLFileSizeKey error:NULL];
+                size += fileSize.unsignedIntegerValue;
+            }
+        }
     }
     return size;
 }
 
 - (NSUInteger)totalCount {
     NSUInteger count = 0;
-    NSDirectoryEnumerator *fileEnumerator = [self.fileManager enumeratorAtPath:self.diskCachePath];
+    NSDirectoryEnumerator<NSString *> *fileEnumerator = [self.fileManager enumeratorAtPath:self.diskCachePath];
```

Comment:
Do we need to change this to URL based as well ?

Reply:
With path, before `@seletor(totalCount)` return , (2196 - 1943) = 253 autorelease objects.

With url,  before `@seletor(totalCount)` return , (1949 - 1943) = 6 autorelease objects.

As a result should fix it.